js/reports.jsx

```diff
@@ -66,61 +64,174 @@ var App = React.createClass({
       </Grid>
       </div>
     )
-  }
+    }
 });
 
 var SearchForm = React.createClass({
-  handleSubmit: function(e) {
-    e.preventDefault();
-    var query = this.refs.searchbox.getDOMNode().value;
-    /* 1. Call the parent's search handler */
-    this.props.onSearch(query);
-  },
-  render: function() {
-    return (
-        <form onSubmit={this.handleSubmit}>
-          <input type="text" id="searchInput" ref="searchbox" />
-          <button type="submit" id="searchSubmit">Search</button>
-        </form>
-    )
-  }
+    handleSubmit: function(e) {
+        e.preventDefault();
+        var runSeriesID = this.refs.searchbox.getDOMNode().value;
+        /* 1. Call the parent's search handler */
+        this.props.getResultData(runSeriesID);
+    },
+    render: function() {
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <input type="text" id="searchInput" ref="searchbox" />
+                <button type="submit" id="searchSubmit">Search</button>
+            </form>
+        )
+    }
+});
+
+var GetProductList = React.createClass({
+    printProducts: function(){
+        if (this.props.productNameData != null) {
+            return this.props.productNameData.objects.map(function (productInfo){
+                return <option id={productInfo.id} value={productInfo.name} label={productInfo.name}>{productInfo.name}</option>;
+            }, this);
+        }
+    },
+
+    componentDidUpdate: function(){
```

Comment:
We can remove this function from code, no need to update state for product name again, will update after dropdownlist on change